pcap-dpdk.c

```diff
@@ -532,10 +534,18 @@ static int dpdk_pre_init(char * ebuf)
 	snprintf(dpdk_cfg_buf,DPDK_CFG_MAX_LEN-1,"%s %s",DPDK_LIB_NAME,ptr_dpdk_cfg);
 	dargv_cnt = parse_dpdk_cfg(dpdk_cfg_buf,dargv);
 	ret = rte_eal_init(dargv_cnt,dargv);
-	// if init successed, we do not need to do it again later.
-	if (ret == 0){
-		is_dpdk_pre_inited = 1;
+	/* Copy the errno */
+	errno = rte_errno;
+
+	if (ret != 0){
+		pcap_fmt_errmsg_for_errno(ebuf, PCAP_ERRBUF_SIZE,
+			EACCES, "dpdk error: %s",
+			DPDK_ERR_RTE_EAL_INIT_FAIL);
+		errno = rte_errno;
```

Comment:
Is there a need to copy the `errno` here if it is being done above on line 538 too?

Reply:
Sorry didn't get that, Tried to improve the comment instead.

Reply:
I'm blind, see it now.